File: 250130/트로미노/tromino.java
import java.util.*;

public class Main {
    static int n;
    static int m;
    static int[][] grid;
    public static void main(String[] args) {
        // Please write your code here.
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();
        m = sc.nextInt();
        grid = new int[n][m];

        for(int i = 0; i < n; i++){
            for(int j = 0; j < m; j++){
                grid[i][j] = sc.nextInt();
            }
        }
        int max = 0;
        int sum = 0;
        for(int i = 0; i < n; i++){
            for(int j = 0; j < m-2; j++){
                sum = grid[i][j] + grid[i][j+1] + grid[i][j+2];
                if(sum > max) max = sum;
            }
        }

        for(int i = 0; i < m; i++){
            for(int j = 0; j < n -2; j++){
                sum = grid[j][i] + grid[j+1][i] + grid[j+2][i];
                if(sum > max) max = sum;
            }
        }

        for(int i = 0; i < n; i++){
            for(int j = 0; j < m; j++){
                if(i + 1 < n && j + 1 < m){
                    sum = grid[i][j] + grid[i+1][j] + grid[i+1][j+1];
                    if(sum > max) max = sum;
                    sum = grid[i][j] + grid[i][j+1] + grid[i+1][j];
                    if(sum > max) max = sum;
                    sum = grid[i][j+1] + grid[i+1][j] + grid[i+1][j+1];
                    if(sum > max) max = sum;
                    sum = grid[i][j] + grid[i][j+1] + grid[i+1][j+1];
                    if(sum > max) max = sum;
                }
            }
        }

        System.out.println(max);   



        
    }
}

File: 260313/트로미노/tromino.py
n, m = map(int, input().split())
grid = [list(map(int, input().split())) for _ in range(n)]

# Please write your code here.

shape = [
    [
        [0, 1, 0],
        [0, 1, 1],
        [0, 0, 0]
    ],
    [
        [0, 0, 0],
        [0, 1, 1],
        [0, 1, 0]
    ],
    [
        [0, 0, 0],
        [1, 1, 0],
        [0, 1, 0]
    ],
    [
        [0, 1, 0],
        [1, 1, 0],
        [0, 0, 0]
    ],
    [
        [0, 1, 0],
        [0, 1, 0],
        [0, 1, 0]
    ],
    [
        [0, 0, 0],
        [1, 1, 1],
        [0, 0, 0]
    ],
]

index= [
    -1, 0, 1
]

max_sum = 0

for r in range(n):
    for c in range(n):
        sum = 0
        for s in range(len(shpae)):
            for x in range(3):
                for y in range(3):
                    xx = r + index[x]
                    yy = c + index[y]
                    if (shape[s][x][y] == 1
                        and 0 <= xx and xx < n  
                        and 0 <= yy and yy <= n):

                        sum += grid[xx][yy]

        
        max_sum = max(max_sum, sum)


p



